x/concentrated-liquidity/pool_test.go

```diff
@@ -190,6 +191,245 @@ func (s *KeeperTestSuite) TestCalcOutAmtGivenIn() {
+				// expectedTick:     87173.8 rounded down https://www.wolframalpha.com/input?i2d=true&i=Log%5B1.0001%2C6105.473934424522538231%5D
+			},
+			tokenIn:       sdk.NewCoin("usdc", sdk.NewInt(10000000000)),
+			tokenOutDenom: "eth",
+			priceLimit:    sdk.NewDec(6106),
+			// expectedTokenIn:  5236545537.865 + 4763454462.135 = 1000000000 usdc
+			// expectedTokenOut: 998587.023 + 821949.120898 = 1820536.143 round down = 1.820536 eth
+			expectedTokenIn:  sdk.NewCoin("usdc", sdk.NewInt(10000000000)),
+			expectedTokenOut: sdk.NewCoin("eth", sdk.NewInt(1820536)),
+			expectedTick:     sdk.NewInt(87173),
+			newLowerPrice:    sdk.NewDec(5501),
+			newUpperPrice:    sdk.NewDec(6250),
+		},
+		// Slippage protection doesn't cause a failure but interrupts early.
+		"single position within one tick, trade completes but slippage protection interrupts trade early: eth -> usdc": {
+			addPositions: func(ctx sdk.Context, poolId uint64) {
+				// add first position
+				_, _, _, err = s.App.ConcentratedLiquidityKeeper.CreatePosition(ctx, poolId, s.TestAccs[0], defaultAmt0, defaultAmt1, sdk.ZeroInt(), sdk.ZeroInt(), lowerTick.Int64(), upperTick.Int64())
+				s.Require().NoError(err)
+			},
+			tokenIn:       sdk.NewCoin("eth", sdk.NewInt(13370)),
+			tokenOutDenom: "usdc",
+			priceLimit:    sdk.NewDec(4994),
+			// params
+			// liquidity: 		 1517818840.967515822610790519
+			// sqrtPriceNext:    70.668238976219012614 which is 4994 https://www.wolframalpha.com/input?i=70.710678118654752440+%2B+42000000+%2F+1517818840.967515822610790519
+			// sqrtPriceCurrent: 70.710678118654752440 which is 5000
+			// expectedTokenIn:  12890.72275 rounded up https://www.wolframalpha.com/input?key=&i=%281517818840.967515822610790519+*+%2870.710678118654752440+-+70.668238976219012614+%29%29+%2F+%2870.710678118654752440+*+70.668238976219012614%29
+			// expectedTokenOut: 64414929.9834 rounded down https://www.wolframalpha.com/input?key=&i=1517818840.967515822610790519+*+%2870.710678118654752440+-+70.668238976219012614%29
+			// expectedTick: 	 85164.2 rounded down https://www.wolframalpha.com/input?key=&i2d=true&i=Log%5B1.0001%2C4994%5D
+			expectedTokenIn:  sdk.NewCoin("eth", sdk.NewInt(12891)),
+			expectedTokenOut: sdk.NewCoin("usdc", sdk.NewInt(64414929)),
+			expectedTick:     sdk.NewInt(85164),
+		},
+		"single position within one tick, trade does not complete due to lack of liquidity: usdc -> eth": {
+			addPositions: func(ctx sdk.Context, poolId uint64) {
+				// add first position
+				_, _, _, err = s.App.ConcentratedLiquidityKeeper.CreatePosition(ctx, poolId, s.TestAccs[0], defaultAmt0, defaultAmt1, sdk.ZeroInt(), sdk.ZeroInt(), lowerTick.Int64(), upperTick.Int64())
+				s.Require().NoError(err)
+			},
+			tokenIn:       sdk.NewCoin("usdc", sdk.NewInt(5300000000)),
+			tokenOutDenom: "eth",
+			priceLimit:    sdk.NewDec(6000),
+			expectErr:     true,
+		},
+		"single position within one tick, trade does not complete due to lack of liquidity: eth -> usdc": {
```

Comment:
Do we have a test over two positions where it does not consume all of the liquidity of the second position?

Also, might be good to have a test over 3 or more positions in the future

Reply:
Added two cases for two positions where we don't not consume all of the liquidity of the second position. Will add the later test in some eventual PR